runtime/src/accounts_db.rs

```diff
@@ -2103,4 +2132,56 @@ pub mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn test_hash_stored_account() {
+        // This test uses some UNSAFE trick to detect most of account's field
+        // addition and deletion without changing the hash code
+
+        const ACCOUNT_DATA_LEN: usize = 3;
+        // the type of InputTuple elements must not contain references;
+        // they should be simple scalars or data blobs
+        type InputTuple = (
+            Slot,
+            StoredMeta,
+            AccountMeta,
+            [u8; ACCOUNT_DATA_LEN],
+            usize, // for StoredAccount::offset
+            Hash,
+        );
+        const INPUT_LEN: usize = std::mem::size_of::<InputTuple>();
+        type InputBlob = [u8; INPUT_LEN];
+        let mut blob: InputBlob = [0u8; INPUT_LEN];
+
+        // spray memory with decreasing counts so that, data layout can be detected.
+        for (i, byte) in blob.iter_mut().enumerate() {
+            *byte = (INPUT_LEN - i) as u8;
+        }
+
+        //UNSAFE: forcibly cast the special byte pattern to actual account fields.
+        let (slot, meta, account_meta, data, offset, hash): InputTuple =
+            unsafe { std::mem::transmute::<InputBlob, InputTuple>(blob) };
+
+        let stored_account = StoredAccount {
+            meta: &meta,
+            account_meta: &account_meta,
+            data: &data,
+            offset,
+            hash: &hash,
+        };
+        let account = stored_account.clone_account();
+        let expected_account_hash =
+            Hash::from_str("GGTsxvxwnMsNfN6yYbBVQaRgvbVLfxeWnGXNyB8iXDyE").unwrap();
```

Comment:
According to [this](https://github.com/solana-labs/solana/pull/7415/files#diff-657d1e7648a9fb74cb1582a9c48beb4bR25) and [that](https://github.com/solana-labs/solana/pull/7415/files#diff-657d1e7648a9fb74cb1582a9c48beb4bR36), this will be really well.

Reply:
Out of pure laziness, I've skipped to calculate this test vector by hand. :p

Reply:
meh! #9917